### PR TITLE
Update boilerplate

### PR DESCRIPTION
**Pull request by davidyuk**
Depends on #126 

I have made a new project using [vuejs-templates/webpack](https://github.com/vuejs-templates/webpack) and synced the created configuration with the configuration of this project.

Most significant changes:
- Webpack 3 instead of Webpack 2
- Jest instead of Mocha, Chai, and Sinon
- `eslint-plugin-vue` instead of `eslint-plugin-html`

Bundle before:

vendor.js parsed size: 7.13mb
app.js parsed size: 391.83kb

After:

vendor.js parsed size: 2.74mb :tada: 
app.js parsed size: 185.18kb

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
   // required to lint *.vue files
   plugins: [
-    'html'
+    'vue'
   ],
   // add your custom rules here
-  'rules': {
-    // allow paren-less arrow functions
-    'arrow-parens': 0,
+  rules: {
```
Comment (muxe) [inline .eslintrc.js]: I think most of these should not be disabled but it's probably good to not also fix all the resulting linter errors in this PR

**Comment by muxe** [comment]
Is build/logo.png needed?

File: .eslintrc.js (hunk 1 above)
Comment (davidyuk) [reply .eslintrc.js]: Yes, that's was an idea, I would like to enable [this rules](https://github.com/aeternity/aepp-identity/pull/127/commits/7a4869c62d86c76c2d22aecd69f11ca9e974bf09) with corresponding fixes in future PR's

**Comment by davidyuk** [comment]
`build/logo.png` is for notifications produced by [node-notifier](https://github.com/aeternity/aepp-identity/pull/127/files#diff-b02e86ddbbbb96a726da148fed31ebf2R98). I'm not sure if it is useful, would like to test.

**Pull request merged**